task4.cpp

```diff
@@ -0,0 +1,83 @@
+#include <iostream>
+#include<fstream>
+
+int strCmp(const char* X, const char* Y) {
+    while (*X) {
+        if (*X != *Y) {
+            break;
+        }
+        X++;
+        Y++;
+    }
+
+    return *(const unsigned char*)X - *(const unsigned char*)Y;
+}
+
+void strCopy(char* destination,const char* source) {
+    if (destination = nullptr) {
+        return;
+    }
+    
+    if (source = nullptr) {
+        return;
+    }
+
+    char* ptr = destination;
+    
+    while (source != '\0') {
+        *destination = *source;
+        destination++;
+        source++;
+    }
+
+    *destination = '\0';
+
+    ptr=destination;
+}
+
+class Secret {
+private:
+    char* task;
+    char* password;
+    int failedLogins=0;
+public:
+    Secret() = default;
+
+    Secret(const char* task, const char* password) {
+        strCopy(this->task, task);
+        strCopy(this->password, password);
+    }
+
+    char* getTask(const char* pwd) {
```

Comment:
methods that don't change the state of the current object must be annotated with const after the method signature